File: client/src/types/Transaction.ts
export type Account = {
  id: number
  name: string
}

export type RegisterAccount = {
  name: string
}

export type Category = {
  id: number
  name: string
}

export type Transaction = {
  id: number
  category: Category
  accountFrom: Account
  accountTo: Account
  date: string
  amount: number
  paid: boolean
  description: string
}

export type RegisterTransaction = {
  categoryId: number
  description: string
  amount: number
  paid: boolean
  date: string
  accountIdFrom: number
  accountIdTo: number
}

export type TransactionsFilter = {
  year?: number
  month?: number
  categoryId?: number
}

File: server/MoneyFlies.Domain/Entities/Transaction.cs
namespace MoneyFlies.Domain.Entities;

public class Transaction : Entity
{
    private Transaction()
    {
    }

    public Transaction(
        decimal amount,
        Category category,
        string description,
        Account from,
        Account to,
        DateOnly date,
        bool paid)
    {
        Amount = amount;
        Category = category;
        Description = description;
        From = from;
        To = to;
        Date = date;
        Paid = paid;
    }

    public int Id { get; private set; }
    public decimal Amount { get; private set; }
    public Category Category { get; private set; }
    public string Description { get; private set; }
    public Account From { get; private set; }
    public Account To { get; private set; }
    public DateOnly Date { get; private set; }
    public bool Paid { get; private set; }

    public void Update(decimal amount, Category category, string description, Account accountFrom, Account accountTo, DateOnly date, bool paid)
    {
        Amount = amount;
        Category = category;
        Description = description;
        From = accountFrom;
        To = accountTo;
        Date = date;
        Paid = paid;
    }
}